tests/deploy/dgd_utils.py

```
@@ -1628,6 +1595,34 @@ def port_forward(
+    def _stop_port_forward(self, port_forward: Any) -> None:
+        """Stop a kr8s port forward and wait for its background thread to exit."""
+        try:
+            port_forward.stop()
+        except RuntimeError as e:
+            # Expected when the pod is terminated while kr8s is cleaning up its
+            # async generator.
+            if "anext()" in str(e) or "already running" in str(e):
+                self._logger.debug(f"Port forward cleanup: {e}")
+            else:
+                self._logger.warning(f"Unexpected error stopping port forward: {e}")
+        except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)